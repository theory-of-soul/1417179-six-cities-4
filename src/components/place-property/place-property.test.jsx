import React from "react";
import renderer from 'react-test-renderer';
import PlaceProperty from "./place-property";

const reviewList = [{
  id: 0,
  rating: 2,
  text: `It was a pain. Wi-Fi was soooo low.`,
  date: new Date(`2020-06-10`),
  userName: `Alex`,
  userIcon: `img/avatar-max.jpg`
}];

const neighbourhoods = [{
  id: 1,
  img: `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/15.jpg`,
  value: 235,
  time: `night`,
  isInBookmark: false,
  rating: 2.4,
  name: `The house among olive `,
  type: `apartment`,
  city: `Dusseldorf`,
  point: [51.237402, 6.797314],
  cityLocation: [51.225402, 6.776314],
  bedrooms: 4,
  description: `I am happy to welcome you to my apartment in the city center! Three words: location, cosy and chic!`,
  goods: [`Laptop friendly workspace`],
  guests: 6,
  host: {
    hostId: 25,
    name: `Angelina`,
    isSuper: true,
    icon: `img/avatar-angelina.jpg`
  },
  images: [`https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/5.jpg`],
  isPremium: false,
}];

describe(`PlaceProperty component snapshot tests`, () => {
  it(`PlaceProperty component premium user`, () => {
    const tree = renderer
      .create(
          <PlaceProperty
            images={[`img/apartment-01.jpg`, `img/apartment-02.jpg`]}
            title={`Beautiful &amp; luxurious studio at great location`}
            description={`A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`}
            isPremium={true}
            type={`room`}
            rating={4}
            rooms={1}
            guests={2}
            price={100}
            goods={[`Wi-Fi`, `Towels`, `Baby seat`]}
            host={{
              icon: `img/avatar-angelina.jpg`,
              name: `Angelina`,
              isSuper: true
            }}
            reviewList={reviewList}
            coordinates={[]}
            cityCoordinates={[]}
            renderMap={() => {}}
            onClickCardTitle={() => {}}
            renderPlaces={() => {}}
            neighbourhoods={neighbourhoods}
          />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
