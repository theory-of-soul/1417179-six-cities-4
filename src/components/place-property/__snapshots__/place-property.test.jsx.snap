// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlaceProperty component snapshot tests PlaceProperty component premium user 1`] = `
<section
  className="property"
>
  <div
    className="property__gallery-container container"
  >
    <div
      className="property__gallery"
    >
      <div
        className="property__image-wrapper"
      >
        <img
          alt="Beautiful &amp; luxurious studio at great location"
          className="property__image"
          src="img/apartment-01.jpg"
        />
      </div>
      <div
        className="property__image-wrapper"
      >
        <img
          alt="Beautiful &amp; luxurious studio at great location"
          className="property__image"
          src="img/apartment-02.jpg"
        />
      </div>
    </div>
  </div>
  <div
    className="property__container container"
  >
    <div
      className="property__wrapper"
    >
      <div
        className="property__mark"
      >
        <span>
          Premium
        </span>
      </div>
      <div
        className="property__name-wrapper"
      >
        <h1
          className="property__name"
        >
          Beautiful &amp; luxurious studio at great location
        </h1>
        <button
          className="property__bookmark-button button"
          type="button"
        >
          <svg
            className="property__bookmark-icon"
            height="33"
            width="31"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        className="property__rating rating"
      >
        <div
          className="property__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "80%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
        <span
          className="property__rating-value rating__value"
        >
          4
        </span>
      </div>
      <ul
        className="property__features"
      >
        <li
          className="property__feature property__feature--entire"
        >
          room
        </li>
        <li
          className="property__feature property__feature--bedrooms"
        >
          1
           Bedrooms
        </li>
        <li
          className="property__feature property__feature--adults"
        >
          Max 
          2
           adults
        </li>
      </ul>
      <div
        className="property__price"
      >
        <b
          className="property__price-value"
        >
          €
          100
        </b>
        <span
          className="property__price-text"
        >
           night
        </span>
      </div>
      <div
        className="property__inside"
      >
        <h2
          className="property__inside-title"
        >
          What's inside
        </h2>
        <ul
          className="property__inside-list"
        >
          <li
            className="property__inside-item"
          >
            Wi-Fi
          </li>
          <li
            className="property__inside-item"
          >
            Towels
          </li>
          <li
            className="property__inside-item"
          >
            Baby seat
          </li>
        </ul>
      </div>
      <div
        className="property__host"
      >
        <h2
          className="property__host-title"
        >
          Meet the host
        </h2>
        <div
          className="property__host-user user"
        >
          <div
            className="property__avatar-wrapper user__avatar-wrapper property__avatar-wrapper--pro"
          >
            <img
              alt="Host avatar"
              className="property__avatar user__avatar"
              height="74"
              src="img/avatar-angelina.jpg"
              width="74"
            />
          </div>
          <span
            className="property__user-name"
          >
            Angelina
          </span>
        </div>
        <div
          className="property__description"
        >
          A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.
        </div>
      </div>
      <section
        className="property__reviews reviews"
      >
        <h2
          className="reviews__title"
        >
          Reviews · 
          <span
            className="reviews__amount"
          >
            1
          </span>
        </h2>
        <ul
          className="reviews__list"
        >
          <li
            className="reviews__item"
          >
            <div
              className="reviews__user user"
            >
              <div
                className="reviews__avatar-wrapper user__avatar-wrapper"
              >
                <img
                  alt="Reviews avatar"
                  className="reviews__avatar user__avatar"
                  height="54"
                  src="img/avatar-max.jpg"
                  width="54"
                />
              </div>
              <span
                className="reviews__user-name"
              >
                Alex
              </span>
            </div>
            <div
              className="reviews__info"
            >
              <div
                className="reviews__rating rating"
              >
                <div
                  className="reviews__stars rating__stars"
                >
                  <span
                    style={
                      Object {
                        "width": "40%",
                      }
                    }
                  />
                  <span
                    className="visually-hidden"
                  >
                    Rating
                  </span>
                </div>
              </div>
              <p
                className="reviews__text"
              >
                It was a pain. Wi-Fi was soooo low.
              </p>
              <time
                className="reviews__time"
                dateTime="2020-06-10"
              >
                June 10, 2020
              </time>
            </div>
          </li>
        </ul>
        <form
          action="#"
          className="reviews__form form"
          method="post"
        >
          <label
            className="reviews__label form__label"
            htmlFor="review"
          >
            Your review
          </label>
          <div
            className="reviews__rating-form form__rating"
          >
            <input
              className="form__rating-input visually-hidden"
              id="5-stars"
              name="rating"
              type="radio"
              value="5"
            />
            <label
              className="reviews__rating-label form__rating-label"
              htmlFor="5-stars"
              title="perfect"
            >
              <svg
                className="form__star-image"
                height="33"
                width="37"
              >
                <use
                  xlinkHref="#icon-star"
                />
              </svg>
            </label>
            <input
              className="form__rating-input visually-hidden"
              id="4-stars"
              name="rating"
              type="radio"
              value="4"
            />
            <label
              className="reviews__rating-label form__rating-label"
              htmlFor="4-stars"
              title="good"
            >
              <svg
                className="form__star-image"
                height="33"
                width="37"
              >
                <use
                  xlinkHref="#icon-star"
                />
              </svg>
            </label>
            <input
              className="form__rating-input visually-hidden"
              id="3-stars"
              name="rating"
              type="radio"
              value="3"
            />
            <label
              className="reviews__rating-label form__rating-label"
              htmlFor="3-stars"
              title="not bad"
            >
              <svg
                className="form__star-image"
                height="33"
                width="37"
              >
                <use
                  xlinkHref="#icon-star"
                />
              </svg>
            </label>
            <input
              className="form__rating-input visually-hidden"
              id="2-stars"
              name="rating"
              type="radio"
              value="2"
            />
            <label
              className="reviews__rating-label form__rating-label"
              htmlFor="2-stars"
              title="badly"
            >
              <svg
                className="form__star-image"
                height="33"
                width="37"
              >
                <use
                  xlinkHref="#icon-star"
                />
              </svg>
            </label>
            <input
              className="form__rating-input visually-hidden"
              id="1-star"
              name="rating"
              type="radio"
              value="1"
            />
            <label
              className="reviews__rating-label form__rating-label"
              htmlFor="1-star"
              title="terribly"
            >
              <svg
                className="form__star-image"
                height="33"
                width="37"
              >
                <use
                  xlinkHref="#icon-star"
                />
              </svg>
            </label>
          </div>
          <textarea
            className="reviews__textarea form__textarea"
            id="review"
            name="review"
            placeholder="Tell how was your stay, what you like and what can be improved"
          />
          <div
            className="reviews__button-wrapper"
          >
            <p
              className="reviews__help"
            >
              To submit review please make sure to set 
              <span
                className="reviews__star"
              >
                rating
              </span>
               and describe your stay with at least 
              <b
                className="reviews__text-amount"
              >
                50 characters
              </b>
              .
            </p>
            <button
              className="reviews__submit form__submit button"
              disabled=""
              type="submit"
            >
              Submit
            </button>
          </div>
        </form>
      </section>
    </div>
  </div>
  <div
    className="container"
  >
    <section
      className="near-places places"
    >
      <h2
        className="near-places__title"
      >
        Other places in the neighbourhood
      </h2>
      <div
        className="near-places__list places__list"
      />
    </section>
  </div>
</section>
`;
